Provide README for the passwordExpiry tool

pbdeuchler opened the pull request.

pbdeuchler committed:
Provide README for the passwordExpiry tool

pbdeuchler committed:
Drastically expand detail and specificity of README for PasswordExpiryWatcher, fix build script to actually build Windows binaries and include ".exe" suffix

pbdeuchler committed:
README formatting changes

beckjake commented on examples/PasswordExpiryWatcher/README.md:
```diff
@@ -0,0 +1,96 @@
+5. `PasswordExpiryWatcher_myos_myarch` is the relevant binary
+	- Some may find this is an unwieldy name to type every time. Feel free to rename to your liking, in fact if this is a tool you plan on running frequently we encourage this for ease of use.
+
+> If you'd like to be able to call this binary from an arbitrary directory you can move it to `/usr/local/bin` on Linux and macOS
```
On Windows, you could add it to C:\Windows\System32\ or anywhere else listed in %PATH%. Dunno if you'd really want to suggest that, but it's an option.

pbdeuchler replied:
My thinking is that people who can connect those dots are probably the only people who will want to actually have the tool in their `PATH`, and if not I'd rather have support walk them through it than have them mess something up thanks to my README

beckjake commented on examples/PasswordExpiryWatcher/README.md:
```diff
@@ -0,0 +1,96 @@
+
+### To Run
+
+To run this tool you will need to use what is called the "command line". On Windows this is a program called "Power Shell", on macOS it is an app called "Terminal". 
```
this is kind of condescending. I'd just say something like `To run this tool you will need to use the command line. On Windows you can use PowerShell and on OSX you can use Terminal`. Also, PowerShell is one word!

beckjake commented on examples/PasswordExpiryWatcher/README.md:
```diff
@@ -0,0 +1,96 @@
+
+Windows:
+
+`./PasswordExpiryWatcher.exe -key=82105124f2979e28273d4e8dd32b2355c5012837 -output=password_expirations.csv`
```
I think you can drop the ".exe" suffix for your commands on Windows, so they can all have the exact same instructions if you want.

beckjake commented on examples/PasswordExpiryWatcher/README.md:
```diff
@@ -0,0 +1,96 @@
+
+> If you have renamed your binary simply replace `PasswordExpiryWatcher` with the new name
+
+> If you installed the binary with the Go toolchain and your `$GOBIN` is in your `$PATH` or `%PATH`, or if you moved the binary to `/usr/local/bin` you can run the above command at any time on your command line excluding the "./"
```
I'm pretty sure it's `%PATH%`, not `%PATH` on Windows. And in PowerShell, I think you need to use something weird like `$env:PATH`? I'm not opening my VM to test that though :)

beckjake commented on examples/PasswordExpiryWatcher/README.md:
```diff
@@ -0,0 +1,96 @@
+> If you installed the binary with the Go toolchain and your `$GOBIN` is in your `$PATH` or `%PATH`, or if you moved the binary to `/usr/local/bin` you can run the above command at any time on your command line excluding the "./"
+
+##### Windows Instructions
+1. Open the program `Power Shell`
```
s/Power Shell/PowerShell/g

beckjake commented on examples/PasswordExpiryWatcher/README.md:
```diff
@@ -0,0 +1,96 @@
+1. Open the program `Power Shell`
+2. Using the `cd` (stands for "Change Directory") command navigate to where you downloaded your binaries
+	- For example, if we downloaded and unzipped the binaries in our `Download` folder we just have to run: `cd Downloads\JumpCloudAPI_Examples_windows_386`. If we downloaded to our desktop the command will probably look something like: `cd Desktop\JumpCloudAPI_Examples_windows_386`
+	- If you used the Go install instructions and your `$GOBIN` is in your `%PATH` you can skip step 2 and go right to 3
```
See above for `%PATH` comment.

beckjake commented on examples/PasswordExpiryWatcher/README.md:
```diff
@@ -0,0 +1,96 @@
+	- `./PasswordExpiryWatcher.exe -key=YOUR_API_KEY_GOES_HERE -output=CSV_FILE_OUTPUT_GOES_HERE`
+	- If you would like your CSV file to go somewhere else besides the current directory make sure you include the _full path_ of the file
+		- Good Example: `-output=C:\Users\MyUser\CSVFiles\password_expirations.csv` 
+		- Bad Example `-output=..\CSVFiles\password_expirations.csv`
```
The fact that this is an issue probably indicates that we're doing something wrong in our code. When tools have this issue it drives me nuts (looking at you, OSX).

pbdeuchler committed:
address CR comments for README, add handling of edge cases for passwordExpiration date

Remove absolute path details since we automatically resolve the absolute path now

beckjake commented:
:+1:

kusold commented:
:+1:
